File: Reclamation.java
package Entities;

import javafx.beans.property.SimpleStringProperty;
import javafx.beans.property.StringProperty;

import java.sql.Date;

public class Reclamation implements Runnable {

    private int id_reclamation;
    private Date date_creation;
    private int id_user;
    private String text;
    private String statut;


    public Reclamation(){};

    public Reclamation(int id_reclamation) {
        this.id_reclamation = id_reclamation;
    }

    public Reclamation(int id_reclamation, String text) {
        this.id_reclamation = id_reclamation;
        this.text = text;
    }

    public Reclamation(int id_reclamation, Date date_creation, int id_user, String text, String statut) {
        this.id_reclamation = id_reclamation;
        this.date_creation = date_creation;
        this.id_user = id_user;
        this.text = text;
 this.statut = statut;    }

    public Reclamation(String text) {

this.text=text;    }

    public String getText() {
        return text;
    }

    public void setText(String text) {
        this.text = text;
    }

    public int getId_reclamation() {
        return id_reclamation;
    }

    public void setId_reclamation(int id_reclamation) {
        this.id_reclamation = id_reclamation;
    }

    public Date getDate_creation() {
        return date_creation;
    }

    public void setDate_creation(Date date_creation) {
        this.date_creation = date_creation;
    }

    public int getId_user() {
        return id_user;
    }

    public void setId_user(int id_user) {
        this.id_user = id_user;
    }


   // public StringProperty statutProperty() { return statut; }
    public String getStatut() {
        return statut;
    }


    public void setStatut(String statut) {
        this.statut=statut;
    }

    @Override
    public String toString() {
        return "Reclamation{" +
                "id_reclamation=" + id_reclamation +
                ", date_creation=" + date_creation +
                ", id_user=" + id_user +
                ", text='" + text + '\'' +
                ", statut='" + statut + '\'' +
                '}';
    }

    @Override
    public void run() {

    }
}

File: Reclamation.php
<?php

namespace App\Entity;

use App\Repository\ReclamationRepository;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;

/**
 * Reclamation
 *
 * @ORM\Table(name="reclamation")
 * @ORM\Entity(repositoryClass=ReclamationRepository::class)
 */
class Reclamation
{
    /**
     * @var int
     *
     * @ORM\Column(name="id_reclamation", type="integer", nullable=false)
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="IDENTITY")
     */
    private $idReclamation;

    /**
     * @var string|null
     *
     * @ORM\Column(name="type", type="string", length=99, nullable=true)
     */
    private $type;

    /**
     * @var \DateTime|null
     *
     * @ORM\Column(name="date_creation", type="date", nullable=true)
     */
    private $dateCreation;

    /**
     * @var \DateTime|null
     *
     * @ORM\Column(name="date_validation", type="date", nullable=true)
     */
    private $dateValidation;

    /**
     * @var int|null
     *
     * @ORM\Column(name="id_user", type="integer", nullable=true)
     */
    private $idUser;

    /**
     * @var string
     *
     * @ORM\Column(name="text", type="string", length=99, nullable=false)
     * @Assert\NotBlank(message="Veuillez remplir ce champs")
     * @Assert\Length(
     *     min="10",
     *     max="100",
     *     minMessage="Doit contenir {{ limit }} caractéres",
     *     maxMessage="Ne doit pas dépasser {{ limit }} caractéres",
     *     )
     */
    private $text;

    /**
     * @var string|null
     *
     * @ORM\Column(name="statut", type="string", length=99, nullable=true)
     */
    private $statut;

    /**
     * @var string|null
     *
     * @ORM\Column(name="screenshot", type="string", length=99, nullable=true)
     */
    private $screenshot;

    /**
     * @var string
     * @Assert\NotBlank(message="Veuillez remplir ce champs")
     * @Assert\Length(
     *     max="30",
     *     maxMessage="Ne doit pas dépasser {{ limit }} caractéres",
     *     )
     */
    private $object;

    public function getIdReclamation(): ?int
    {
        return $this->idReclamation;
    }

    public function getType(): ?string
    {
        return $this->type;
    }

    public function setType(?string $type): self
    {
        $this->type = $type;

        return $this;
    }

    public function getDateCreation(): ?\DateTimeInterface
    {
        return $this->dateCreation;
    }

    public function setDateCreation(?\DateTimeInterface $dateCreation): self
    {
        $this->dateCreation = $dateCreation;

        return $this;
    }

    public function getDateValidation(): ?\DateTimeInterface
    {
        return $this->dateValidation;
    }

    public function setDateValidation(?\DateTimeInterface $dateValidation): self
    {
        $this->dateValidation = $dateValidation;

        return $this;
    }

    public function getIdUser(): ?int
    {
        return $this->idUser;
    }

    public function setIdUser(?int $idUser): self
    {
        $this->idUser = $idUser;

        return $this;
    }

    public function getText(): ?string
    {
        return $this->text;
    }

    public function setText(string $text): self
    {
        $this->text = $text;

        return $this;
    }

    public function getStatut(): ?string
    {
        return $this->statut;
    }

    public function setStatut(?string $statut): self
    {
        $this->statut = $statut;

        return $this;
    }

    public function getScreenshot(): ?string
    {
        return $this->screenshot;
    }

    public function setScreenshot(?string $screenshot): self
    {
        $this->screenshot = $screenshot;

        return $this;
    }

    public function getObject(): ?string
    {
        return $this->object;
    }

    public function setObject(string $object): self
    {
        $this->object = $object;

        return $this;
    }


}
